lib/luacompletion.c

```diff
@@ -0,0 +1,88 @@
+/*
+* SPDX-FileCopyrightText: (c) 2023-2024 Ring Zero Desenvolvimento de Software LTDA
+* SPDX-License-Identifier: MIT OR GPL-2.0-only
+*/
+
+#define pr_fmt(fmt) KBUILD_MODNAME ": " fmt
+#include <linux/kernel.h>
+#include <linux/module.h>
+#include <linux/completion.h>
+
+#include <lua.h>
+#include <lualib.h>
+#include <lauxlib.h>
+
+#include <lunatik.h>
+
+LUNATIK_OBJECTCHECKER(luacompletion_check, struct completion *);
+
+static int luacompletion_complete(lua_State *L)
+{
+	struct completion *completion = luacompletion_check(L, 1);
+
+	complete(completion);
+	return 0;
+}
+
+static int luacompletion_wait(lua_State *L)
+{
+	struct completion *completion = luacompletion_check(L, 1);
+	lua_Integer timeout = luaL_optinteger(L, 2, MAX_SCHEDULE_TIMEOUT);
+	unsigned long timeout_jiffies = msecs_to_jiffies((unsigned long)timeout);
+	unsigned long ret;
+
+	ret = wait_for_completion_interruptible_timeout(completion, timeout_jiffies);
+	if (ret == 0 || ret == -ERESTARTSYS) {
+		lua_pushnil(L);
+		lua_pushstring(L, ret == 0 ? "timeout" : "interrupted");
```

Comment:
I think "interrupt" is more concise.. but I have no strong feelings about this...feel free to choose =)

Reply:
it could be. I hadn't considered it, but it looks good. will change.